File: Models/Product.cs
namespace RetailOrderingWebsite.Models;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Price { get; set; }

    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public int SellerId { get; set; }
    public Seller Seller { get; set; } = null!;

    public int Stock { get; set; }
}

File: frontend/src/app/shared/models/product.ts
export interface Category {
  id: number;
  name: string;
}

export interface Seller {
  id: number;
  name: string;
  email: string;
  phone: string;
  address: string;
}

export interface Product {
  id: number;
  name: string;
  price: number;
  stock: number;
  category: Category;
  seller: Seller;
}
